### ISSUE TITLE
CLI flags/configuration and examples for Linux GPU training

### ISSUE THREAD

**Issue by bbrowning** [instructlab/instructlab#647]
The exact settings that need to change to successfully train with a Linux GPU can vary quite a lot by system. AMD cards vs Nvidia cards, memory available on the card, age of card, etc.

A list of the settings I've found so far that a user may want to tweak to get training working at all or to make tradeoffs in overall training speed versus resources used, nonexhaustive:

- device(s) to use
- fp16 vs bf16 precision
- quantitization, specifically using 4bit BitsAndBytes vs not
- gradient accumulation steps or disabled entirely
- gradient checkpointing enabled/disabled
- per-device training batch size
- distributed training across multiple GPUs/CPUs (may be out of scope for just config, as that's more work to setup)

Some of these are configurable by CLI flags today. Can we expose all the needed parameters via CLI flags? Do we need configuration files? Which of these are also applicable to other lab commands, such as serve, generate, test, convert?

However we expose the necessary configuration, a list of example configuration/flags to use for different setups would be nice. Show people the things to tweak to lower memory usage at the expense of speed. Perhaps some guidance on the options needed to reduce GPU memory required under popular thresholds, like 8GB, 16GB, 24GB.

My assumption here is a goal would be to give people enough knobs to turn that they can get training going on their machine without having to change the actual Python code to do it. Perhaps others disagree with that assumption? All opinions are welcome!

**Comment by nathan-weinberg** [comment]
@bbrowning is this issue still relevant or can it be closed?

**Comment by bbrowning** [comment]
We still don't expose all the necessary knobs so that users can get the most out of the simple Linux training in this repo without editing the source code, so this issue is still relevant in that sense. How much work we want to do to enable this depends on how far we want to push training here versus focusing on instructlab/training repo for Linux training work.

**Comment by nathan-weinberg** [comment]
@RobotSail @Maxusmusti @JamesKunstle same question for you fellas here

**Comment by bbrowning** [comment]
Given all the changes since this was opened, I don't think it's still exactly relevant today. Closing this as outdated, and we can open specific issues based on the current CLI workflow as needed.
